In addition looking for plugins named lib*, look for libZeroc* to
match the renamed Ice libraries for compatibility with xorg-libice's
libICE.*.

diff -ru Ice-3.5b.orig/cpp/src/Ice/DynamicLibrary.cpp Ice-3.5b/cpp/src/Ice/DynamicLibrary.cpp
--- Ice-3.5b.orig/cpp/src/Ice/DynamicLibrary.cpp	2012-12-12 13:15:12.000000000 -0800
+++ Ice-3.5b/cpp/src/Ice/DynamicLibrary.cpp	2012-12-13 19:04:42.000000000 -0800
@@ -148,10 +148,15 @@
 
     lib += ".dll";
 #elif defined(__APPLE__)
+    // Look for libZeroc... to handle the MacPorts
+    // s/libIce/libZerocIce/ rename.
+    string libZeroc = lib;
     lib += "lib" + libName;
+    libZeroc += "libZeroc" + libName;
     if(!version.empty())
     {
         lib += "." + version;
+        libZeroc += "." + version;
     }
 #elif defined(__hpux)
     lib += "lib" + libName;
@@ -185,7 +190,10 @@
     //
     if(!load(lib + ".dylib"))
     {
-        string errMsg = _err;
+      string errMsg = _err;
+      if(!load(libZeroc + ".dylib"))
+      { 
+        errMsg += "; " + _err;
         if(!load(lib + ".so"))
         {
             errMsg += "; " + _err;
@@ -195,7 +203,8 @@
                 return 0;
             }
         }
-        _err = "";
+      }
+      _err = "";
     }
 #else
     if(!load(lib))
